### PR TITLE
Reflect active section

### PR DESCRIPTION
**Pull request by Kikobeats**
TODO

- [x] Add a way to retrieve the active section.
- [x] Reflect active section at `Aside`.
- [x] Reflect active section on GitHub URL at `Nav`.

### CODE REVIEW

File: components/aside.js [hunk 1]
```diff
@@ -23,6 +23,14 @@ const Aside = styled(Box)`
     text-overflow: ellipsis;
   }
 
+  ${props =>
+    props.hash &&
+    `
+  a[href="${props.hash}"] {
```
Comment (Kikobeats) [inline components/aside.js]: Am I a hacker?

File: components/hook/hash-change.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { useEffect } from 'react'
+
+const getHash = () => window.location.hash
+
+let currentHash
+
+export default fn => {
```
Comment (axelhzf) [inline components/hook/hash-change.js]: If you are writing a hook it should be name `useSomething`

File: components/hook/hash-change.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  useEffect(() => {
+    window.document.addEventListener(
+      'scrollStart',
```
Comment (Kikobeats) [inline components/hook/hash-change.js]: This event listener approximation is just a temporal thing until determinate a better way using `Router` provided my Next.js

File: components/hook/hash-change.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+      () => {
+        const hash = getHash()
+        if (hash !== currentHash) {
+          currentHash = hash
+          fn(hash)
+        }
+      },
+      false
+    )
```
Comment (axelhzf) [inline components/hook/hash-change.js]: You need to remove the event listener on unmount

```
useEffect(() => {
   function cb() {

   }
  window.document.addEventListener('scrollStart', cb)
  return () => windows.document.removeEventListener('scrollStart', cb)
});


```

File: components/hook/hash-change.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import { useEffect } from 'react'
+
+const getHash = () => window.location.hash
+
+let currentHash
```
Comment (Kikobeats) [inline components/hook/hash-change.js]: Although this is working like a charm I'm not sure if a more logical workflow could be using `React.Context` API.

**Pull request merged**